src/java.base/share/classes/java/util/FormatProcessor.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package java.util;
+
+import java.lang.invoke.*;
+import java.lang.invoke.MethodHandles.Lookup;
+import java.lang.template.*;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import jdk.internal.javac.PreviewFeature;
+
+/**
+ * This {@linkplain ValidatingProcessor yemplate processor} constructs a String
+ * result using {@link Formatter}. Unlike {@link Formatter}, FormatProcessor uses the value from
+ * the embedded expression that follows immediately after the
+ * <a href="../util/Formatter.html#syntax">format specifier</a>.
+ * StringTemplate expressions without a preceeding specifier, use "%s" by
+ * default. Example:
+ * {@snippet :
+ * int x = 10;
+ * int y = 20;
+ * String result = FMT."%05d\{x} + %05d\{y} = %05d\{x + y}";
+ * }
+ * result is: <code>00010 + 00020 = 00030</code>
+ *
+ * @implNote When used in conjunction with a compiler generated {@link
+ * StringTemplate} this {@link ValidatingProcessor} will use the format
+ * specifiers in the fragments and types of the values in the value list
+ * to produce a more performant formatter.
+ *
+ * @implSpec Since, values are in situ, argument indexing is unsupported.
```

Comment:
Please avoid Latin phrases like "in situ" in the javadoc.

Reply:
Changing